src/Stage/InferTypes/GenerateEquations.elm

```diff
@@ -412,3 +456,165 @@ generateArgumentUsageEquations argumentId usages =
     List.map
         (Typed.getType >> equals argumentType)
         usages
+
+
+generatePatternEquations : Id -> Typed.LocatedPattern -> ( List TypeEquation, Id )
+generatePatternEquations currentId located =
+    let
+        ( pttrn, type_ ) =
+            Located.unwrap located
+    in
+    case pttrn of
+        Typed.PAnything ->
+            -- we can't make any assumptions here
+            ( [], currentId )
+
+        Typed.PVar _ ->
+            -- we can't make any assumptions here
+            ( [], currentId )
+
+        Typed.PRecord _ ->
+            -- we can't make any assumptions here
+            ( [], currentId )
```

Comment:
I wonder if we can or can not. I guess we don't have enough "infrastructure" to be able to say "it's a record but I'm not gonna say more about what's inside"... I guess we can let this one be for now?

Reply:
You are right, we can say "It's a record and have at least these keys". I will take a look and see if I have the knowledge to express it or, at least, improve the comment.